src/zope/publisher/browser.py

```diff
@@ -295,16 +296,14 @@ def processInputs(self):
         else:
             fp = None
 
-        # If 'QUERY_STRING' is not present in self._environ
-        # FieldStorage will try to get it from sys.argv[1]
-        # which is not what we need.
-        if 'QUERY_STRING' not in self._environ:
-            self._environ['QUERY_STRING'] = ''
+        # We could simply not parse QUERY_STRING if it's absent, but this
+        # provides slightly better doctest-compatibility.
+        self._environ.setdefault('QUERY_STRING', '')
 
         # The Python 2.6 cgi module mixes the query string and POST values
         # together.  We do not want this.
         env = self._environ
-        if self.method == 'POST' and self._environ['QUERY_STRING']:
+        if self.method == 'POST' and 'QUERY_STRING' in self._environ:
```

Comment:
(Note: my comments here are not meant to imply that I prefer this approach to #55)

The comment above about what Python 2.6 cgi module does could be updated to refer to what our vendored copy of the Python 3.8 cgi module does.